executor/builder.go

```diff
@@ -4544,6 +4583,32 @@ func (builder *dataReaderBuilder) buildIndexLookUpReaderForIndexJoin(ctx context
 		err = e.open(ctx)
 		return e, err
 	}
+
+	is := v.IndexPlans[0].(*plannercore.PhysicalIndexScan)
+	ts := v.TablePlans[0].(*plannercore.PhysicalTableScan)
+	if is.Index.Global {
+		tmp, ok := builder.is.TableByID(ts.Table.ID)
+		if !ok {
+			return nil, err
+		}
```

Comment:
Forgot set err here. By the way, I think no need check ok here, just like L4612.

Also please fix other similar codes.

Reply:
Yes👍🏻，You are right. But I find other places in this method also not set err. When the builder.err should be set?

Reply:
I think you could change it to `return nil, infoschema.ErrTableNotExists` here.

Reply:
The err without set maybe always equal to `nil`.

Reply:
resolved